app/services/process_steps.rb

```diff
@@ -0,0 +1,40 @@
+class ProcessSteps
+  def initialize(steps:, step:, adjust:)
+    @steps = steps
+    @step = step
+    @adjust = adjust
+  end
+
+  def current_step_number
+    current_step = if defined?(@step)
```

Comment:
When is @step undefined? I think never; it comes from `step_current` which defaults to `:job_location`.

Reply:
I don't think it would be – `@step` is initialised in the constructor.

Also, undefined instance variables are always `nil` so you don't need to check whether they're defined.

Reply:
Also the `params[:controller]` below won't work anyway as this is not a controller

Reply:
i think this might explain my failing system spec, thanks!